mtags/src/main/scala-2/scala/meta/internal/pc/InferredMethodProvider.scala

```diff
@@ -0,0 +1,520 @@
+                Ident(containingMethod),
+                arguments
+              ) :: _ if nonExistent.isErrorTyped =>
+            makeEditsForApplyWithUnknownName(
+              arguments,
+              containingMethod,
+              errorMethod,
+              nonExistent
+            )
+          // <<nonExistent>>(param1, param2)
+          // val a: Int = <<nonExistent>>(param1, param2)
+          case (_: Ident) :: Apply(
+                containing @ Ident(
+                  nonExistent
+                ),
+                arguments
+              ) :: _ if containing.isErrorTyped =>
+            signature(
+              name = nonExistent.toString(),
+              paramsString = Option(argumentsString(arguments).getOrElse("")),
+              retType = None,
+              postProcess = identity,
+              position = None
+            )
+          // List(1, 2, 3).map(<<nonExistent>>)
+          // List((1, 2)).map{case (x,y) => <<nonExistent>>(x,y)}
+          case (_: Ident) :: Apply(
+                Ident(containingMethod),
+                arguments
+              ) :: _ =>
+            makeEditsForListApply(arguments, containingMethod, errorMethod)
+          // List(1,2,3).map(myFn)
+          case (_: Ident) :: Apply(
+                Select(
+                  Apply(
+                    Ident(_),
+                    arguments
+                  ),
+                  _
+                ),
+                _
+              ) :: _ =>
+            signature(
+              name = errorMethod.name.toString(),
+              paramsString =
+                Option(argumentsString(arguments.take(1)).getOrElse("")),
+              retType = None,
+              postProcess = identity,
+              position = None
+            )
+          // val list = List(1,2,3)
+          // list.map(nonExistent)
+          case (Ident(_)) :: Apply(
+                Select(Ident(argumentList), _),
+                _ :: Nil
+              ) :: _ =>
+            makeEditsForListApplyWithoutArgs(argumentList, errorMethod)
+          case _ => unimplemented("simple-case")
+        }
+      case errorMethod: Select =>
+        lastVisitedParentTrees match {
+          // class X{}
+          // val x: X = ???
+          // x.nonExistent(1,true,"string")
+          case Select(Ident(container), _) :: Apply(
+                _,
+                arguments
+              ) :: _ =>
+            makeEditsMethodObject(Option(arguments), container, errorMethod)
+          // X.<<nonExistent>>
+          case Select(Ident(container), _) :: _ =>
+            makeEditsMethodObject(Option.empty, container, errorMethod)
+          // X.<<nonExistent>>
+          case Select(Select(_, container), _) :: _ =>
+            makeEditsMethodObject(Option.empty, container, errorMethod)
+          case _ =>
+            unimplemented("other-than-method")
+        }
+      case _ =>
+        unimplemented("other-than-ident-or-select")
```

Comment:
We could add better explanations here and use the DisplayableError which would then show up for the users.

Reply:
this code is just a reference to find it back in code, I will make the logger into the DisplayableError.
I am unsure what would be a better explanation than saying "the case x is not currently supported for the action".
There could be many cases unsupported for these bits  (it would take time to list them and not sure how that would help a user who is likely just disappointed things don't work here)

Reply:
I think it's enough to make it into a sentence, so that it's clear what is not supported.

Reply:
okay I applied the changes.